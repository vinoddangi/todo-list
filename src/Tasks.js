import {
  Box,
  Typography,
  Container as MuiContainer,
  Stack,
  TextField,
  Button,
  Grid,
} from '@mui/material';
import React from 'react';
import { styled } from '@mui/material/styles';
import { HomeOutlined } from '@mui/icons-material';
import { addToList } from './store/entity/entitySlice';
import { useDispatch } from 'react-redux';
import { TaskList } from './TaskList';

function Tasks() {
  const dispatch = useDispatch();
  const Container = styled(MuiContainer)(({ theme }) => ({
    backgroundColor: theme.palette.primary.light,
    border: `1px solid ${theme.palette.primary.dark}`,
    color: 'white',
  }));
  function addTodo() {
    dispatch(addToList({ property: 'todos', value: { text: 'test', favorite: true } }));
  }
  return (
    <Container maxWidth="sm" sx={{ padding: '16px 0' }}>
      <Grid container spacing={2}>
        <Grid item xs={12}>
          <Stack direction="row" spacing={2} alignItems={'center'}>
            <HomeOutlined />
            <Typography variant="h6" gutterBottom component="div">
              Tasks
            </Typography>
          </Stack>
        </Grid>
        <Grid item xs={12}>
          <TaskList />
        </Grid>
        <Grid item xs={9}>
          <TextField
            id="todoText"
            placeholder="Add todo"
            variant="outlined"
            fullWidth
            sx={{ backgroundColor: 'white' }}
            size="small"
          />
        </Grid>

        <Grid item xs={3}>
          <Button variant="contained" onClick={addTodo}>
            Add Todo
          </Button>
        </Grid>
      </Grid>
    </Container>
  );
}
export { Tasks };
