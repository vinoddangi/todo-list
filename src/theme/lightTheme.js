import { createTheme, darkScrollbar } from '@mui/material';
const lightTheme = createTheme({
  components: {
    MuiCssBaseline: {
      styleOverrides: {
        body: darkScrollbar(),
      },
    },
    MuiAppBar: {
      styleOverrides: {
        root: {
          minHeight: '32px',
          '@media(min-width:600px)': {
            minHeight: '48px',
          },
          width: '100%',
          '@media(min-width:600px)': {
            width: '60%',
          },
        },
      },
    },
  },
});
export { lightTheme };
