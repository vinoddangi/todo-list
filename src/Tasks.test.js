import renderer from 'react-test-renderer';
import { Tasks } from './Tasks';
import React from 'react';
import { ThemeProvider } from '@mui/material';
import { lightTheme } from './theme/lightTheme';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('Given Tasks', () => {
  let tree;
  beforeEach(() => {
    const component = renderer.create(
      <ThemeProvider theme={lightTheme}>
        <Tasks />
      </ThemeProvider>
    );
    tree = component.toJSON();
  });

  test('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
