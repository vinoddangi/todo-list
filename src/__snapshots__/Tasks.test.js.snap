// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Tasks renders correctly 1`] = `
<div
  alignitems="stretch"
  className="MuiContainer-root MuiContainer-maxWidthSm css-73ez2z-MuiContainer-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="css-1yo1jo6-MuiStack-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HomeOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 5.69 5 4.5V18h-2v-6H9v6H7v-7.81l5-4.5M12 3 2 12h3v8h6v-6h2v6h6v-8h3L12 3z"
          />
        </svg>
        <div
          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
        >
          Tasks
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="css-17n6opp"
      >
        <div
          className="css-1iz4iln"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary css-1kcggdq-MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Search…"
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-tv4j1g-MuiGrid-root"
    >
      <ul
        className="MuiList-root MuiList-padding css-62wvau-MuiList-root"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-md-9 css-1kp9ysn-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-12l054h-MuiFormControl-root-MuiTextField-root"
        onKeyDown={[Function]}
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="todoText"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Add todo"
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-md-3 css-g04lnj-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Add Todo
      </button>
    </div>
  </div>
</div>
`;
